### ISSUE TITLE
Handle all macOS file events without dropping VFS

### ISSUE THREAD

**Issue by lptr** [gradle/gradle#11836]
Currently we drop the VFS contents if some unrecognized event is reported on `macOS`. We can handle most if not all of these events, and prevent the VFS from dropping.

---
cc: @gradle/build-cache

**Comment by lptr** [comment]
Looks like we are changing the owner here:

```
~~~~ Unknown event 0x24000 for /Users/lptr/Workspace/gradle/gradle/subprojects/platform-play/build/tmp/test files/unknown-test-class/emtnt/daemon/6.3-20200204230000+0000
```

That's:

```
kFSEventStreamEventFlagItemIsDir       = 0x20000
kFSEventStreamEventFlagItemChangeOwner = 0x04000
```

This forces a VFS drop and masks problems in `CachedPathSensitivityIntegrationTest` with the `clean` task not updating the VFS.

**Comment by lptr** [comment]
https://github.com/gradle/native-platform/pull/85 has been merged and `master` is now using it since #12194 has been merged.
